scapy/utils.py

```diff
@@ -1365,12 +1365,18 @@ def import_hexcap():
 
 
 @conf.commands.register
-def wireshark(pktlist):
+def wireshark(pktlist, text_mode=False):
     """Run wireshark on a list of packets"""
     f = get_temp_file()
     wrpcap(f, pktlist)
-    with ContextManagerSubprocess("wireshark()", conf.prog.wireshark):
-        subprocess.Popen([conf.prog.wireshark, "-r", f])
+    if text_mode:
+        prog = "tshark"
+        args = [conf.prog.tshark, "-V", "-r"]
+    else:
+        prog = "wireshark"
```

Comment:
Certainly not! Please use `conf.prog.xxx`.

Reply:
See just below :)

`prog` variable here is only used in the context manager as the name of the util that fails if an error happens

Reply:
Oh of course yes!